### ISSUE TITLE
Add a wire-level integration test

### ISSUE THREAD

**Issue by stephenh** [stephenh/ts-proto#8]
I.e. stand up a java/go/something server and make calls to it.

**Comment by ToonvanStrijp** [comment]
@stephenh We have this now for the `nestjs` integration but we could add more implementations such as `java` or `go` but I'm not sure if this is necessary?

**Comment by stephenh** [comment]
Yeah, that's true, but I'd like to keep this for doing wire-level tests of the non-NestJS integrations, i.e. the `Message.encode/decode` methods and `Method.fromJSON/toJSON`.

**Comment by timostamm** [comment]
Google protobuf has conformance tests. They are using an executable that runs a test suite against any implementation. It uses a similar pattern to protoc plugins, with a test request and test response. The suite can test binary, json and text format. All official implementations are tested against this suite, so it should be very well suited to make sure that ts-proto is fully compatible.

https://github.com/protocolbuffers/protobuf/tree/master/conformance

Of course there is a gotcha. There is no binary release for the conformance tests. 
I have zero experience with C++ and have not been able to build the project yet.

**Comment by stephenh** [comment]
Ah nice, good find! That does look useful to eventually have it running against ts-proto.

**Comment by nikelborm** [comment]
Well... Someone already did

https://github.com/bufbuild/protobuf-conformance#results
